cypress/integration/smokeTests/sharedTests/auth-tests.ts

```diff
@@ -38,10 +38,7 @@ function deleteTool() {
     storeToken();
     cy.server();
     cy.route('delete', '**/containers/**').as('containers');
-    cy.wait(2000); // hardcoded 2s wait is least flaky option right now, revisit in future
-    cy.contains('#deregisterButton', 'Delete').should('be.enabled');
-    cy.contains('#deregisterButton', 'Delete').click();
-    // cy.get('#deregisterButton').click();
+    cy.contains('#deregisterButton', 'Delete').should('be.visible').click();
```

Comment:
would this be even safer if we also included a "be.enabled" (per the replaced code)?  not sure, genuinely curious. :)

Reply:
TLDR: As long as the correct DOM element is passed to `.click()`, there shouldn't be a difference between `be.enabled` and `be.visible`, because `.click()` checks for visibility and being enabled anyways.

Ideally, `.should('be.visible')` wouldn't be required at all. This is because the Cypress `.click()` action checks for [actionability](https://docs.cypress.io/guides/core-concepts/interacting-with-elements#Actionability), which includes making sure the element is enabled, visible, etc... But, for an unknown reason, the command `cy.contains('#deregisterButton', 'Delete').click();` would fail, while `cy.contains('#deregisterButton', 'Delete').should('be.visible').click();` succeeds.

Just hypothesizing, but I expect the DOM element yielded from `cy.contains('#deregisterButton', 'Delete').click();` can be grabbed too quickly, and that DOM element is never updated to the actionable state (perhaps the button is reloaded after a few moments, and the reloaded button is the one that becomes visible?). By changing the line to `cy.contains('#deregisterButton', 'Delete').should('be.visible').click();`, Cypress will wait for a visible button to be in the DOM before passing it on to `.click()`, and this seems to fix the issue I was having locally.